osquery/tables/ipmi/ipmi_info.cpp

```diff
@@ -0,0 +1,1297 @@
+  r["firmware_minor_version"] = INTEGER(ipmi_mc_minor_fw_revision(mc));
+  r["ipmi_major_version"] = INTEGER(ipmi_mc_major_version(mc));
+  r["ipmi_minor_version"] = INTEGER(ipmi_mc_minor_version(mc));
+  r["iana_manufacturer_id"] = INTEGER(ipmi_mc_manufacturer_id(mc));
+  r["product_id"] = INTEGER(ipmi_mc_product_id(mc));
+  r["provides_device_sdrs"] = INTEGER(ipmi_mc_provides_device_sdrs(mc));
+  r["chassis_support"] = INTEGER(ipmi_mc_chassis_support(mc));
+  r["bridge_support"] = INTEGER(ipmi_mc_bridge_support(mc));
+  r["ipmb_event_generator_support"] =
+      INTEGER(ipmi_mc_ipmb_event_generator_support(mc));
+  r["ipmb_event_reciever_support"] =
+      INTEGER(ipmi_mc_ipmb_event_receiver_support(mc));
+  r["fru_inventory_support"] = INTEGER(ipmi_mc_fru_inventory_support(mc));
+  r["sel_device_support"] = INTEGER(ipmi_mc_sel_device_support(mc));
+  r["sdr_respository_support"] = INTEGER(ipmi_mc_sdr_repository_support(mc));
+  r["sensor_device_support"] = INTEGER(ipmi_mc_sensor_device_support(mc));
+  r["is_active"] = INTEGER(ipmi_mc_is_active(mc));
+
+  unsigned char guid = 0;
+  int rv = ipmi_mc_get_guid(mc, &guid);
+  switch (rv) {
+  case ENOSYS:
+    r["guid"] = "not available";
+    break;
+
+  case 0:
+    r["guid"] = std::string(reinterpret_cast<const char*>(&guid), 16);
+    break;
+
+  default:
+    LOG(ERROR) << "Unexpected error retrieving MC GUID: " << strerror(rv);
+  }
+
+  IPMIClient* c = reinterpret_cast<IPMIClient*>(data);
+  c->insertRowsQueue(r["name"] + "-" + r["device_id"], r);
+}
+
+IPMIClient& IPMIClient::get() {
+  static IPMIClient c;
+  return c;
+}
+
+bool IPMIClient::isUp() {
+  return up_.load();
+}
+
+IPMIClient::~IPMIClient() {
+  up_.store(false);
+}
+
+IPMIClient::IPMIClient()
+    : up_(false), domain_(nullptr), os_hnd_(nullptr, kFreeOSHandle), lanCh_(0) {
+  LOG(WARNING) << "First time initialization of OpenIPMI client.  This could "
+                  "take a few minutes";
+
+  std::unique_ptr<os_handler_t, std::function<void(os_handler_t*&)>> tempHandle(
+      ipmi_posix_setup_os_handler(), kFreeOSHandle);
+  if (tempHandle.get() == nullptr) {
+    LOG(ERROR) << "Could not allocate posix handler with ipmi_smi_setup_con";
+    return;
+  }
+  os_hnd_.swap(tempHandle);
+
+  TLOG << "Setting OpenIPMI log handler";
+  os_hnd_.get()->set_log_handler(os_hnd_.get(), ipmiLoggerCB);
+
+  TLOG << "Initializing OpenIPMI client";
+  int rv = ipmi_init(os_hnd_.get());
+  if (rv != 0) {
+    LOG(ERROR) << "IPMI initialization failed: " << strerror(rv);
+    return;
+  }
+
+  TLOG << "Setting up OpenIPMI connection";
+  // Only support 1 IPMI connection for now..
+  rv = ipmi_smi_setup_con(0, os_hnd_.get(), NULL, &con_);
+  if (rv != 0) {
+    LOG(WARNING) << "Error setting up SMI connection: " << strerror(rv);
+    return;
+  }
+
+  TLOG << "Opening IPMI domain";
+  if (ipmi_open_domain(
+          "", &con_, 1, NULL, NULL, IPMIFullyUpCB, this, NULL, 0, NULL) != 0) {
```

Comment:
nit: For consistency with some of your other functions, would you mind pulling this out and assigning to `rv`? 
```
rv = ipmi_open_domain("", &con_, 1, NULL, NULL, IPMIFullyUpCB, this, NULL, 0, NULL);
if (rv != 0) {
...
```